src/test/scala/it/pagopa/interop/agreementprocess/AgreementActivationSpec.scala

```diff
@@ -76,12 +76,26 @@ class AgreementActivationSpec extends AnyWordSpecLike with SpecHelper with Scala
         verifiedAttributes = Seq(VerifiedAttribute(verAttr1), VerifiedAttribute(verAttr2)),
         suspendedByConsumer = None,
         suspendedByProducer = Some(false),
-        suspendedByPlatform = Some(false)
+        suspendedByPlatform = Some(false),
+        stamps = SpecData.activationStamps
       )
 
       mockAgreementRetrieve(agreement)
       mockAgreementsRetrieve(Nil)
       mockEServiceRetrieve(eService.id, eService)
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockAttributeManagementServiceRetrieve(SpecData.clientAttribute(UUID.randomUUID()))
+      mockPDFCreatorCreate
+      mockFileManagerWrite
+      mockAgreementContract
+      mockPartyManagementRetrieve(SpecData.institution(UUID.randomUUID()))
+      mockPartyManagementRetrieve(SpecData.institution(UUID.randomUUID()))
+      mockUserRegistryRetrieve(SpecData.userResource("a", "b", "c"))
+      mockUserRegistryRetrieve(SpecData.userResource("d", "e", "f"))
```

Comment:
may I suggest to wrap these under a `mockContractCreation` function?
IMO it's easier to understand and maintain